portal-wire-protocol.md

```diff
@@ -0,0 +1,163 @@
+
+```
+message_id := 0x01
+type       := request
+sedes      := Container(enr_seq: uint64, data_radius: uint256, custom_payload: ByteList)
+```
+
+* `enr_seq`: The node's current sequence number of their ENR record.
+* `data_radius`: The nodes current maximum radius for data stored by this node.
+* `custom_payload`: Custom payload dependant on the network.
+
+
+### Pong (0x02)
+
+Response message to Ping(0x01)
+
+```
+message_id := 0x02
+type       := response
+sedes      := Container(enr_seq: uint64, data_radius: uint256)
+```
+
+* `enr_seq`: The node's current sequence number of their ENR record.
+* `data_radius`: The nodes current maximum radius for data stored by this node.
+* `custom_payload`: Custom payload dependant on the network.
+
+### Find Nodes (0x03)
+
+Request nodes from the peer's routing table at the given logarithmic distances.  The distance of `0` indicates a request for the peer's own ENR record.
+
+```
+message_id := 0x03
+type       := request
+sedes      := Container(distances: List[uint16, max_length=256])
+```
+
+* `distances`: a list of distances for which the node is requesting ENR records for.
+    * Each distance **MUST** be within the inclusive range `[0, 256]`
+    * Each distance in the list **MUST** be unique.
+
+### Nodes (0x04)
+
+Response message to FindNodes(0x03).
+
+```
+message_id := 0x04
+type       := response
+sedes      := Container(total: uint8, enrs: List[ByteList, max_length=32])
+```
+
+* `total`: The total number of `Nodes` response messages being sent.
+* `enrs`: List of bytestrings, each of which is an RLP encoded ENR record.
+    * Individual ENR records **MUST** correspond to one of the requested distances.
+    * It is invalid to return multiple ENR records for the same `node_id`.
+
+> Note: If the number of ENR records cannot be encoded into a single message, then they should be sent back using multiple messages, with the `total` field representing the total number of messages that are being sent.
+
+### Find Content (0x05)
+
+Request either the data payload for a specific piece of content on the network, **or** ENR records of nodes that are closer to the requested content.
+
+```
+message_id := 0x05
+type       := request
+sedes      := Container(content_key: ByteList)
+```
+
+* `content_key`: The key for the content being requested. The encoding of `content_key` is dependant on the network.
+
+### Found Content (0x06)
+
+Response message to Find Content (0x05).
+
+This message can contain either a uTP connection ID, a list of ENRs or the
+requested content.
+
+```
+message_id := 0x06
+type       := response
+sedes      := Union[connection-id: Bytes2, content: Bytes, enrs: List[ByteList, 32]]
+```
+
+* `connection_id`: Connection ID to set up a uTP stream to transmit the requested data.
+    * Connection ID values should be randomly generated.
```

Comment:
Do we want the "should" here to be italicized or bolded?

Reply:
We should probably be consistent everywhere and either use caps or caps+bold everywhere.